src/assets/src/hooks/useGoogleAnalytics.ts

```diff
@@ -1,22 +1,72 @@
 import { useState, useEffect } from 'react';
 import GoogleAnalytics from 'react-ga4';
 import { useLocation } from 'react-router-dom';
+import { useOneTrust } from './useOneTrust';
+
 
 export const useGoogleAnalytics = (googleAnalyticsId?: string, debug?: boolean) => {
     let location = useLocation();
+
     const [initialized, setInitialized] = useState(false);
     const [previousPage, setPreviousPage] = useState(null as string | null);
 
     if (googleAnalyticsId && !initialized) {
+        GoogleAnalytics.gtag("consent", "default", {
+            ad_storage: "denied",
+            analytics_storage: "denied",
+            functionality_storage: "denied",
+            personalization_storage: "denied",
+            ad_user_data: "denied",
+            ad_personalization: "denied",
+            wait_for_update: 500
```

Comment:
Optional: Make the `denied` as const and add as object value  here. So if need to change only one place to change

Reply:
added an enum for this purpose